src/main/java/ru/skypro/homework/service/impl/AuthServiceImpl.java

```diff
@@ -16,23 +18,36 @@ public class AuthServiceImpl implements AuthService {
 
     private final UserDetailsManager manager;
     private final PasswordEncoder encoder;
+    private final CustomUserDetailsService customUserDetailsService;
+    private final UserRepository userRepository;
 
     static ArrayList<String> userLog = new ArrayList<>();
 
-    public AuthServiceImpl(UserDetailsManager manager,
-                           PasswordEncoder passwordEncoder) {
+    //public AuthServiceImpl(UserDetailsManager manager, PasswordEncoder encoder, UserRepository userRepository) {
+    //    this.manager = manager;
+    //    this.encoder = encoder;
+    //    this.userRepository = userRepository;
+    //}
+
+
+    public AuthServiceImpl(UserDetailsManager manager, PasswordEncoder encoder, CustomUserDetailsService customUserDetailsService, UserRepository userRepository) {
         this.manager = manager;
-        this.encoder = passwordEncoder;
+        this.encoder = encoder;
+        this.customUserDetailsService = customUserDetailsService;
+        this.userRepository = userRepository;
     }
 
     @Override
     public boolean login(String userName, String password) {
-        if (!manager.userExists(userName)) {
+        //if (!manager.userExists(userName)) {
+        if (userRepository.findByUsername(userName) == null) {
             return false;
         } else {
-            UserDetails userDetails = manager.loadUserByUsername(userName);
+            //UserDetails userDetails = manager.loadUserByUsername(userName);
+            UserDetails userDetails = customUserDetailsService.loadUserByUsername(userName);
             userLog.add(userName);
-            return encoder.matches(password, userDetails.getPassword());
+            return encoder.matches(password, encoder.encode(userDetails.getPassword())); //Тут немного не уверен. ???
```

Comment:
```suggestion
            return encoder.matches(password, userDetails.getPassword()); //Тут немного не уверен. ???
```
у тебя уже в userDetails пароль хэшированный, повторно это делать не нужно